src/arm/64/sad.S

```diff
@@ -77,10 +77,9 @@ sad_rect 4, 16
 .macro horizontal_add_16x8
         uaddlp          v0.4s,   v0.8h
         uaddlp          v0.2d,   v0.4s
-        dup             d1,  v0.d[0]
-        dup             d0,  v0.d[1]
+        ext             v1.16b,  v0.16b,  v0.16b,  #8
         add             v0.2s,   v1.2s,   v0.2s
-        umov            w0,  v0.s[0]
+        fmov            w0,  s0
```

Comment:
Why `fmov`?

Reply:
Referencing "Cortex-A57 Software Optimization Guide" the throughput of this instruction is twice that of the alternative although the result is equivalent. There is a comment to this effect in the XNU strlen implementation, "`umov.b` would be more natural, but requires 2 µops." Since this is the tail of the function, the effect is barely measurable.